apps/financial-aid/web-osk/pages/api/auth/[...nextauth].ts

```diff
@@ -0,0 +1,90 @@
+import NextAuth, { Callbacks } from 'next-auth'
+import Providers from 'next-auth/providers'
+import {
+  GenericObject,
+  NextApiRequest,
+  NextApiResponse,
+} from 'next-auth/_utils'
+import { uuid } from 'uuidv4'
+import {
+  AuthSession,
+  AuthUser,
+  checkExpiry,
+  decodeToken,
+  getSession,
+  handleSignIn,
+  identityServerConfig,
+  refreshAccessToken,
+  RolesRule,
+} from '@island.is/financial-aid/shared/lib'
+
+const providers = [
+  Providers.IdentityServer4({
+    id: identityServerConfig.id,
+    name: identityServerConfig.name,
+    scope: identityServerConfig.scope,
+    clientId: identityServerConfig.clientId,
+    domain: process.env.IDENTITYSERVER_DOMAIN ?? '',
+    clientSecret: process.env.IDENTITYSERVER_SECRET ?? '',
+  }),
+]
+
+const callbacks: Callbacks = {
+  signIn: signIn,
+  jwt: jwt,
+  session: session,
+}
+
+async function signIn(
+  user: AuthUser,
+  account: GenericObject,
+  profile: GenericObject,
+): Promise<boolean> {
+  return handleSignIn(user, account, profile)
+}
+
+async function jwt(token: GenericObject, user: AuthUser) {
+  if (user) {
+    token = {
+      nationalId: user.nationalId,
+      name: user.name,
+      accessToken: user.accessToken,
+      refreshToken: user.refreshToken,
+      idToken: user.idToken,
+      isRefreshTokenExpired: false,
+      folder: token.folder ?? uuid(),
+      service: RolesRule.OSK,
+    }
+  }
+
+  if (checkExpiry(token.accessToken, token.isRefreshTokenExpired)) {
+    try {
+      const [accessToken, refreshToken] = await refreshAccessToken(
```

Comment:
Is this working now?

Reply:
Going to test it properly today, but it has been a while since I got the refresh error 😊